torch/csrc/distributed/rpc/init.cpp

```diff
@@ -414,6 +410,19 @@ If the future completes with an error, an exception is thrown.
             `datetime.timedelta` instance indicating the RPC timeout.
       )");
 
+  module.def(
+      "enable_gil_profiling",
+      [](bool flag) {
+        RpcAgent::getCurrentRpcAgent()->enableGILProfiling(flag);
+      },
+      R"(
+    Set whether GIL wait times should be enabled or not. This incurs a slight
+    overhead cost. Default is disabled for performance reasons.
+
+    Arguments:
+        flag (bool): True to set GIL profiling, False to disable.
+      )");
```

Comment:
the old `enable_gil_profiling` used to release GIL, but no longer so in the new API. I guess the new version is the right approach, as it would be better to holding GIL for super light method to avoid ctx switching cost.